From 7523a6c11b578a16a546e6909cd81842ed10e017 Mon Sep 17 00:00:00 2001
From: Iain Lane <iainl@gnome.org>
Date: Fri, 23 Nov 2018 10:32:34 +0000
Subject: [PATCH 41/55] Only execute fully executable desktop files

For us, this means

  - metadata::trusted true
  - UNIX executable bit set
---
 fileItem.js | 16 +++++++---------
 1 file changed, 7 insertions(+), 9 deletions(-)

diff --git a/fileItem.js b/fileItem.js
index 5deb846..203b94a 100644
--- a/fileItem.js
+++ b/fileItem.js
@@ -402,16 +402,14 @@ var FileItem = class {
             return;
         }
 
-        if (this._attributeCanExecute && !this._isDirectory) {
-            if (this._isDesktopFile) {
-                this._desktopFile.launch_uris_as_manager([], null, GLib.SpawnFlags.SEARCH_PATH, null, null);
-                return;
-            }
-
-            if (!this._execLine)
-                return;
+        if (this.trustedDesktopFile) {
+            this._desktopFile.launch_uris_as_manager([], null, GLib.SpawnFlags.SEARCH_PATH, null, null);
+            return;
+        }
 
-            Util.spawnCommandLine(this._execLine);
+        if (this._attributeCanExecute && !this._isDirectory && !this._isDesktopFile) {
+            if (this._execLine)
+                Util.spawnCommandLine(this._execLine);
             return;
         }
 
-- 
2.20.0

